src/main/java/org/trigon/utilities/mirrorable/Mirrorable.java

```diff
@@ -22,26 +23,26 @@ public abstract class Mirrorable<T> {
     private static boolean IS_RED_ALLIANCE = isRedAlliance();
 
     protected final T nonMirroredObject, mirroredObject;
-    protected final boolean mirrorWhenRedAlliance;
-
-    static {
-        UPDATE_ALLIANCE_TIMER.start();
-
-        new Trigger(() -> UPDATE_ALLIANCE_TIMER.advanceIfElapsed(0.5)).onTrue(
-                new InstantCommand(() -> IS_RED_ALLIANCE = notCachedIsRedAlliance())
-        );
-    }
+    protected final boolean shouldMirrorWhenRedAlliance;
 
     /**
      * Creates a new mirrorable object.
      *
-     * @param nonMirroredObject     the object when the robot is on the blue alliance, or the non-mirrored object
-     * @param mirrorWhenRedAlliance whether to mirror the object when the robot is on the red alliance
+     * @param nonMirroredObject           the object when the robot is on the blue alliance, or the non-mirrored object
+     * @param shouldMirrorWhenRedAlliance whether the object should be mirrored when the robot is on the red alliance
      */
-    protected Mirrorable(T nonMirroredObject, boolean mirrorWhenRedAlliance) {
+    protected Mirrorable(T nonMirroredObject, boolean shouldMirrorWhenRedAlliance) {
         this.nonMirroredObject = nonMirroredObject;
         this.mirroredObject = mirror(nonMirroredObject);
-        this.mirrorWhenRedAlliance = mirrorWhenRedAlliance;
+        this.shouldMirrorWhenRedAlliance = shouldMirrorWhenRedAlliance;
+    }
+
+    /**
+     * Initializes the mirrorable object. This should be called once in robot container's constructor.
+     */
+    public static void init() {
+        UPDATE_ALLIANCE_TIMER.start();
+        new Trigger(() -> UPDATE_ALLIANCE_TIMER.advanceIfElapsed(0.5)).onTrue(getUpdateAllianceCommand());
```

Comment:
While this might work, I'm not sure if this is the cleanest solution, or if this fully makes sense. I think this stems from another problem at line 23. Try fixing that and seeing if that helps.